tests/libres_tests/res/job_queue/test_job_queue_manager.py

```diff
@@ -36,152 +34,158 @@ def dummy_exit_callback(args):
+    command used to submit jobs to LSF"""
+    os.chdir(tmpdir)
+    os.putenv("PATH", os.getcwd() + ":" + os.getenv("PATH"))
+    driver = Driver(driver_type=QueueDriverEnum.LSF_DRIVER, max_running=1)
+
+    script = Path(DUMMY_CONFIG["job_script"])
+    script.write_text(SIMPLE_SCRIPT, encoding="utf-8")
+    script.chmod(stat.S_IRWXU | stat.S_IRWXO | stat.S_IRWXG)
+
+    bsub = Path("bsub")
+    bsub.write_text(MOCK_BSUB, encoding="utf-8")
+    bsub.chmod(stat.S_IRWXU | stat.S_IRWXO | stat.S_IRWXG)
+
+    job_id = 0
+    num_cpus = 4
+    os.mkdir(DUMMY_CONFIG["run_path"].format(job_id))
+    job = JobQueueNode(
+        job_script=DUMMY_CONFIG["job_script"],
+        job_name=DUMMY_CONFIG["job_name"].format(job_id),
+        run_path=os.path.realpath(DUMMY_CONFIG["run_path"].format(job_id)),
+        num_cpu=num_cpus,
+        status_file="STATUS",
+        ok_file="OK",
+        exit_file="ERROR",
+        done_callback_function=DUMMY_CONFIG["ok_callback"],
+        exit_callback_function=DUMMY_CONFIG["exit_callback"],
+        callback_arguments=[
+            {"job_number": job_id},
+            Path(DUMMY_CONFIG["run_path"].format(job_id)).resolve(),
+        ],
+    )
+
+    pool_sema = BoundedSemaphore(value=2)
+    job.run(driver, pool_sema)
+    job.stop()
+    job.wait_for()
+
+    bsub_argv: List[str] = Path("test.out").read_text(encoding="utf-8").split()
+
+    found_cpu_arg = False
+    for arg_i, arg in enumerate(bsub_argv):
+        if arg == "-n":
+            # Check that the driver submitted the correct number
+            # of cpus:
+            assert bsub_argv[arg_i + 1] == str(num_cpus)
+            found_cpu_arg = True
+
+    assert found_cpu_arg is True
+
+
+def test_execute_queue(tmpdir):
+    os.chdir(tmpdir)
+    job_queue = create_local_queue(SIMPLE_SCRIPT)
+    manager = JobQueueManager(job_queue)
+    manager.execute_queue()
+
+    assert job_queue.isRunning is False
+
+    for job in job_queue.job_list:
+        assert (Path(job.run_path) / "OK").read_text(encoding="utf-8") == "success"
+
+
+@pytest.mark.parametrize("max_submit_num", [1, 2])
+def test_max_submit_reached(tmpdir, max_submit_num):
+    """ "Check that the JobQueueManager will submit exactly the maximum number of
+    resubmissions in the case of scripts that fail."""
+    os.chdir(tmpdir)
+    num_realizations = 2
+    job_queue = create_local_queue(
+        FAILING_SCRIPT, max_submit=max_submit_num, num_realizations=num_realizations
+    )
+
+    manager = JobQueueManager(job_queue)
+    manager.execute_queue()
+
+    assert (
+        Path("one_byte_pr_invocation").stat().st_size
+        == max_submit_num * num_realizations
+    )
+
+    assert manager.isRunning() is False
+
+    for job in job_queue.job_list:
+        # one for every realization
+        assert job.status == JobStatusType.JOB_QUEUE_FAILED
+        assert job.submit_attempt == job_queue.max_submit
+
+
+@pytest.mark.parametrize("max_submit_num", [1, 2])
```

Comment:
Don't know the impact of the test, but `max_submit_num` was set 5 in the old test. So maybe like `@pytest.mark.parametrize("max_submit_num", [3, 4, 5])`